fix: connection dropped for refresh-data [TCTC-6839]

Sanix-Darker opened the pull request:
## WHAT

`connection dropped` on remote file (SFTP).
This is not a "new issue", my proposition on this PR is to increase the timeout, the retry and
suppress the .close() error and let see on live prod how it behave.
+ add more logging information.

## HOW

- feat: increase timeout values and the retry amount
- fix: suppress the .close error for None type ?
- chore: add more logs while accessing the remote SFTP file.

Sanix-Darker committed:
feat: increase timeout values and the retry amount

Sanix-Darker committed:
fix: suppress the .close error for None type ?

WilliamGorge commented on peakina/io/ftp/ftp_utils.py:
```diff
@@ -16,6 +16,8 @@
 import paramiko
 
 FTP_SCHEMES = ["ftp", "ftps", "sftp"]
+_DEFAULT_MAX_TIMEOUT = 10
```
tu peux spécifier l'unité dans la constante?
```suggestion
_DEFAULT_MAX_TIMEOUT_SECONDS = 10
```

Sanix-Darker replied:
https://github.com/toucantoco/peakina/pull/600/commits/e4e4a5d

WilliamGorge reviewed (changes requested):
Merci

Sanix-Darker committed:
fix: add second in the const name

WilliamGorge commented:
on merge pas tant qu'on n'est pas sur que ça fix le souci du client

Sanix-Darker committed:
feat: log file list from the directory to see what's inside

WilliamGorge commented on peakina/io/ftp/ftp_utils.py:
```diff
@@ -188,9 +189,32 @@ def ftp_open(url: str, retry: int = _DEFAULT_MAX_RETRY) -> IO[bytes]:  # type: i
+            # TODO: remove this after the debuging is done !
+            if isinstance(e, FileNotFoundError):
+                logging.getLogger(__name__).warning(
+                    url.split("/")[-1] + " not found !"
```
si on veut garder ce code: j'aurais mis toute l'url

Sanix-Darker replied:
ok, jusqu'au path ducoup, parcequ'il y'a des creds dedans 😬 
j'update cela !

WilliamGorge replied:
ah!

WilliamGorge replied:
dans ce cas tant pis

Sanix-Darker replied:
No worries : https://github.com/toucantoco/peakina/pull/600/commits/752f10a

WilliamGorge commented on peakina/io/ftp/ftp_utils.py:
```diff
@@ -188,9 +189,32 @@ def ftp_open(url: str, retry: int = _DEFAULT_MAX_RETRY) -> IO[bytes]:  # type: i
+            # if this occurs, we need to see what's actually inside that dir
+            # by listing maxi 15 entries
+            # TODO: remove this after the debuging is done !
+            if isinstance(e, FileNotFoundError):
```
si on veut garder ce code: tu ajouter un gros try/except pour ce bloc au cas ou y'a une erreur? pour éviter de crash dans le fallback

Sanix-Darker replied:
alright !

WilliamGorge commented on peakina/io/ftp/ftp_utils.py:
```diff
@@ -188,9 +189,32 @@ def ftp_open(url: str, retry: int = _DEFAULT_MAX_RETRY) -> IO[bytes]:  # type: i
-            sleep(sleep_time)
+            # if this occurs, we need to see what's actually inside that dir
+            # by listing maxi 15 entries
+            # TODO: remove this after the debuging is done !
```
pour moi on peut le garder... à voir

WilliamGorge commented on peakina/io/ftp/ftp_utils.py:
```diff
@@ -185,30 +185,34 @@ def _open(url: str) -> IO[bytes]:
+                    )
+                    file_listed = True
+                except Exception as e:
+                    pass # fail in silent
```
non faut log ou raise

Sanix-Darker committed:
chore: adding more details on the logging

WilliamGorge commented on peakina/io/ftp/ftp_utils.py:
```diff
@@ -177,14 +184,42 @@ def _open(url: str) -> IO[bytes]:
+
+                    files_available = ", ".join(ftp_listdir(modified_url)[:15])
+                    # we list only 15 as maximum
+                    log.warning(f"List of files : ({files_available})")
```
ça serait plus simple et lisible d'avoir qu'un seul log

Sanix-Darker replied:
Updates here : https://github.com/toucantoco/peakina/pull/600/commits/a72fc9c

Sanix-Darker committed:
fix: default timeout was originally at 60s,

but for each ftp-client it was defined at 3s,
Setting it to 10s is too small when we have a long list of files that could takes too much time to resolve,
the value should be increased, in this commit, am tripling the default fomr FTPS class that was at 60s to 180s.

Later on, we should find a way to set a FF that will forward that
extremly HUGE value to the lib itself.

Sanix-Darker committed:
fix: catch SSHException to trigger the retry on connection dropped

Sanix-Darker committed:
chore: more cleaning, one log with details on the file we're not seeing

Sanix-Darker committed:
chore: added the entry on CHANGELOG.md

WilliamGorge commented on peakina/io/ftp/ftp_utils.py:
```diff
@@ -16,6 +17,8 @@
 import paramiko
 
 FTP_SCHEMES = ["ftp", "ftps", "sftp"]
+_DEFAULT_MAX_TIMEOUT_SECONDS = 30
+_DEFAULT_MAX_RETRY = 7
```
tu augmente quand même le timeout et les retry?

Sanix-Darker replied:
le retry etait 4
le timeout etait 3

Tu suggeres que je remettes ces valeurs ?
Je me disais que ces valeurs ne sont pas deconantes non plus.

et  si tu regardes la valeur du timeout par defaut de la calsse FTP, il etait a 60s  : https://github.com/toucantoco/peakina/blob/main/peakina/io/ftp/ftp_utils.py#L27

WilliamGorge commented on peakina/io/ftp/ftp_utils.py:
```diff
@@ -117,13 +122,15 @@ def sftp_client(params: ParseResult) -> Generator[tuple[paramiko.SFTPClient, str
-        ssh_client.close()
+        # In cae of Exception, we don't want to raise it
+        with suppress(AttributeError):
+            ssh_client.close()
```
je loggerait quand même le cas ou on arrive pas à close du coup

Sanix-Darker replied:
https://github.com/toucantoco/peakina/pull/600/commits/006ebab

Sanix-Darker committed:
chore: add log in AttributeError for the .close()
